test/connectors/parsers/GetPenaltyDetailsParserSpec.scala

```diff
@@ -89,6 +142,21 @@ class GetPenaltyDetailsParserSpec extends AnyWordSpec with Matchers with LogCapt
         val result = GetPenaltyDetailsParser.GetPenaltyDetailsReads.read("GET", "/", mockOKHttpResponseWithInvalidBody)
         result.isLeft shouldBe true
       }
+
+      "there is a missing clearing date for posted LPP1" in {
+        val result = GetPenaltyDetailsParser.GetPenaltyDetailsReads.read("GET", "/", mockOKHttpResponseWithMissingClearingDateForOnePostedLPP1Body)
+
+        result.isRight shouldBe true
+        result.toOption.get.asInstanceOf[GetPenaltyDetailsSuccessResponse].penaltyDetails.latePaymentPenalty.get.details.get.filter(_.penaltyChargeReference.contains("123456789")).head.principalChargeLatestClearing.get shouldBe lpp2Details.principalChargeLatestClearing.get
+      }
+
+      "there are missing clearing dates for posted LPP1s" in {
```

Comment:
Could there be a test where we have an LPP1 that already has a clearing date set so:
- LPP1 posted (no clearing date - gets updated)
- LPP2 posted (clearing date)
- LPP1 posted (clearing date, different principal charge)
Just to make sure that it doesn't alter unrelated LPPs?